azpubsub/src/main/scala/com/microsoft/azpubsub/kafka/admin/AzPubSubAclCommand.scala

```diff
@@ -0,0 +1,176 @@
+        filteredAcls.foreach(x => filteredAclOperations.add(x.operation()))
+        if (resource._1.resourceType() == ResourceType.TOPIC) {
+          if (producerAclOperations.subsetOf(filteredAclOperations)) {
+            strayAcls = strayAcls.filterNot(x => (x.host() == "*" && x.permissionType() == AclPermissionType.ALLOW && producerAclOperations.contains(x.operation())))
+            var modifiedAcl = new AzPubSubAccessControlEntry(principal, "*", AclOperation.ANY, AclPermissionType.ALLOW, "PRODUCER")
+            producerConsumerGroupAclMap(resource._1).add(modifiedAcl)
+          }
+          if (consumerAclOperations.subsetOf(filteredAclOperations)) {
+            strayAcls = strayAcls.filterNot(x => (x.host() == "*" && x.permissionType() == AclPermissionType.ALLOW && consumerAclOperations.contains(x.operation())))
+            var modifiedAcl = new AzPubSubAccessControlEntry(principal, "*", AclOperation.ANY, AclPermissionType.ALLOW, "CONSUMER")
+            producerConsumerGroupAclMap(resource._1).add(modifiedAcl)
+          }
+        }
+        else if (resource._1.resourceType() == ResourceType.GROUP) {
+          if (groupAclOperations.subsetOf(filteredAclOperations)) {
+            strayAcls = strayAcls.filterNot(x => (x.host() == "*" && x.permissionType() == AclPermissionType.ALLOW && groupAclOperations.contains(x.operation())))
+            var modifiedAcl = new AzPubSubAccessControlEntry(principal, "*", AclOperation.ANY, AclPermissionType.ALLOW, "GROUP")
+            producerConsumerGroupAclMap(resource._1).add(modifiedAcl)
+          }
+        }
+        strayAcls.foreach(acl => {
+          var modifiedAcl = new AzPubSubAccessControlEntry(principal, acl.host(), acl.operation(), acl.permissionType(), "NONE")
+          producerConsumerGroupAclMap(resource._1).add(modifiedAcl)
+        })
+      }}
+    })
+    return producerConsumerGroupAclMap
+  }
+}
+
+class AzPubSubAdminClientService(opts: AzPubSubAclCommandOptions) extends AdminClientService(opts) {
+
+  override def printAcls(filters: Set[ResourcePatternFilter], listPrincipals: Set[KafkaPrincipal], resourceToAcls: Map[ResourcePattern, Set[AccessControlEntry]]): Unit = {
+    if (!opts.options.has(opts.outputAsProducerConsumerOpt)) {
+      super.printAcls(filters, listPrincipals, resourceToAcls)
+      return
+    }
+
+    if (listPrincipals.isEmpty) {
+      var producerConsumerAclMap = AzPubSubAclCommand.aclToProducerConsumerMapping(resourceToAcls);
+      outputAsJson(producerConsumerAclMap)
+    }
+    else {
+      val allPrincipalsFilteredResourceToAcls = resourceToAcls.mapValues(acls =>
+        acls.filterNot(acl => listPrincipals.forall(
+          principal => !principal.toString.equals(acl.principal)))).filter(entry => entry._2.nonEmpty)
+      var producerConsumerAclMap = AzPubSubAclCommand.aclToProducerConsumerMapping(allPrincipalsFilteredResourceToAcls)
```

Comment:
we can move aclToProducerConsumerMapping & associated methods to AzPubSubAdminClientService class itself as it is not used outside